prov/efa/src/efa_env.c

```diff
@@ -228,8 +228,8 @@ void efa_env_define()
 			"Enables fork support and disables internal usage of huge pages. Has no effect on kernels which set copy-on-fork for registered pages, generally 5.13 and later. (Default: false)");
 	fi_param_define(&efa_prov, "runt_size", FI_PARAM_INT,
 			"The maximum number of bytes that will be eagerly sent by inflight messages uses runting read message protocol (Default 307200).");
-	fi_param_define(&efa_prov, "use_sm2", FI_PARAM_BOOL,
-			"Use the experimental shared memory provider SM2 for intra node communication.");
+	fi_param_define(&efa_prov, "intranode_provider", FI_PARAM_STRING,
+			"The name of the peer provider that EFA should offload intra-node communications to (Default shm).");
```

Comment:
you would have to add check here for allowed values. Currently only shm and sm2 are allowed.